### PR TITLE
Wip/gi meson

### PR DESCRIPTION
**Pull request by guruDanny67**
Build and install gobject-introspection, using a modified version of wip/tingping/meson from the original gnome repository.

Tested with python 3.5.3 (with and without a virtualenv) and with the latest python 3.6.3

This is not the final release (we should, at least, use the original gnome repository or a release) but it's here to get some feedback and see if everything is ok.

### CODE REVIEW

File: gvsbuild/groups.py [hunk 1]
```diff
@@ -31,6 +31,7 @@ def __init__(self):
             dependencies = [
                 'cmake',
                 'meson',
+                'msys',
```
Comment (anthrotype) [inline gvsbuild/groups.py]: excuse me for my ignorance, but isn't msys2 (required by gvsbuild) supposed to be the new and improved version of msys?

Comment (anthrotype) [reply gvsbuild/groups.py]: forget my comment sorry, of course you do mean msys2 here.
by the way, thanks again for working on this!

Comment (guruDanny67) [reply gvsbuild/groups.py]: It's msys2 but all over the script is named msys (e.g. def exec_msys(...), opts.msys_dir, ...) so I keep the same name.

Probably is correct and better call the project, in the tools, and in the dependencies, msys2.

**Comment by TingPing** [comment]
Since g-i checks for 2.55 and 2.54 is packaged here maybe it makes sense to patch that until we get caught up? (It should be reasonably safe)

**Comment by nacho** [comment]
Should we get this in with the sugestion from @TingPing ?

**Comment by guruDanny67** [comment]
Yes, I used in one of my project build in python 3.6.3 this gobject-introspection linked to the glib 2.52.3 and seems to work correctly so keep a little patch should be no problem.

I also put in one on my branch the creation of the gir/typelib for the other gtk libs (atk, gtk3) that can be put in the pull request.

It's probably better to merge all the modification for the build in separate commits and, at the end, prepare a new pull request with only the gobject-introspection stuff.

What do you think ?

Thanks in advance

**Comment by nacho** [comment]
Sounds good to me

**Comment by guruDanny67** [comment]
All is made, a little bit better, in #208 so we close this.

**Pull request closed without merging**